wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -326,37 +343,42 @@ private MachineImpl createMachine(MachineConfigImpl machineConfig,
                                                     null);
 
         createMachineLogsDir(machineId);
-        final LineConsumer machineLogger = getMachineLogger(machineId, getMachineChannels(machine.getConfig().getName(),
-                                                                                          machine.getWorkspaceId(),
-                                                                                          machine.getEnvName())
-                .getOutput());
+        final LineConsumer machineLogger = getMachineLogger(machineId, outputConsumer);
 
         try {
+            machineRegistry.addMachine(machine);
```

Comment:
should we add machine in machineRegistry only in case of success ? (I mean adding it when instance is successfully created, and not before and then check exception and remove)